src/common/Network/CommonActorControl.h

```diff
@@ -77,6 +77,7 @@ enum ActorControlType : uint16_t
     TreasureScreenMsg = 0x57,
     SetOwnerId = 0x59,
     ItemRepairMsg = 0x5C,
+    BluActionLearn = 0x63, // Triggers the "Action Learned!" animation and tells which action is learned. Any action ID may be used
```

Comment:
I wonder if this is exclusive to blue then. Might just be for any action/job?

Reply:
When BLU unlocks a spell, it sends a regular ToggleActionUnlock and 0x63 (0x63 is actually sent for each Blue Mages in the party since they all learn the spell at the same time, but oddly enough the animation isn't visible on other players) So I assume it's exclusive to BLU.

I also believe it's only used for visual purposes and doesn't actually unlock anything. I tried to inject the packet with an empty Blue Magic Spellbook and all it did was play the "Action Learned" animation and put the "newly acquired" green dot on the corresponding learned spell, but it didn't actually unlock it, even within the book.
![blulearn](https://user-images.githubusercontent.com/25123300/56294277-019c0180-612b-11e9-9fe5-13574760d07f.PNG)
![blulearn2](https://user-images.githubusercontent.com/25123300/56294279-019c0180-612b-11e9-8ee3-afd1502894ba.PNG)

Reply:
Be that as it may, as you've mentioned, you could use it on any action as a visual confirmation.
I find it unlikely ( granted not impossible ) that something with such a low id in the list would be exclusive to something so recently added.

Reply:
I guess we can agree to put a comment saying //verification needed on it ;)

Reply:
Done! :P

I should have been clearer from the start, but this is the animation I was talking about:
![blulearn3](https://user-images.githubusercontent.com/25123300/56305181-08813f00-6140-11e9-82b5-cf197dedcbbf.PNG)

However, you're totally right that the fact it's possible to only display the "You learn Song of Torment!" part without the big "Action Learned!", and that it accepts any actionId and not only BLU ones makes it usable for all jobs.

Reply:
I think they might then use this for job quests? those show the little notification with "You learn x" don't they?

Reply:
yeah pretty sure this is like `ActionLearnNotification` and `param2` is like the display variant or something. would need to check an old executable but don't have em at work, but my guess is that they've just added that to the same actor control as an additional param

Reply:
the actor control opcode is also pretty low too which is indicative of the fact that it's probably been around for a while as well